combinations/hash.tsv

```diff
@@ -7,4 +7,6 @@ tensorflow=1.4.0,tensorboard=0.4.0rc3,matplotlib==2.1.0,numpy=1.13.3,scipy=0.19.
+_r-mutex=1.0.0,ca-certificates=2018.4.16,gettext=0.19.8.1,icu=56.1,libgcc-ng=7.2.0,libgfortran=3.0.0,libgfortran-ng=7.2.0,libstdcxx-ng=7.2.0,nomkl=2.0,openjdk=8.0.121,pixman=0.34.0,readline=6.2,tk=8.5.19,xz=5.0.5,boost=1.57.0,findutils=4.6.0,jpeg=9c,libffi=3.2.1,libgcc=7.2.0,libiconv=1.15,libopenblas=0.2.20,ncurses=6.0,openblas=0.2.19,openssl=1.0.2o,pcre=8.41,sqlite=3.19.3,zlib=1.2.11,bedtools=2.26.0,blas=1.1,glib=2.51.4,libpng=1.6.35,libtiff=4.0.6,libxml2=2.9.3,python=3.5.0,certifi=2018.4.16,cromwell=0.34,freetype=2.6.3,numpy=1.11.3,pyparsing=2.2.0,pytz=2018.5,sip=4.18,six=1.11.0,cycler=0.10.0,fontconfig=2.11.1,python-dateutil=2.7.3,scipy=1.0.0,setuptools=40.0.0,cairo=1.14.6,wheel=0.31.1,harfbuzz=1.3.4,pip=18.0,pango=1.40.3,qt=4.8.7,r-base=3.2.2,pyqt=4.11.4,matplotlib=1.5.1,idr=2.0.4.2,bioconductor-biocgenerics=0.18.0,bioconductor-s4vectors=0.8.11,bioconductor-iranges=2.4.8,bioconductor-genomeinfodb=1.6.3,bioconductor-zlibbioc=1.16.0,bioconductor-xvector=0.10.0,bioconductor-biostrings=2.38.4,bioconductor-genomicranges=1.22.4,r=3.2.2,r-bitops=1.0_6,r-boot=1.3_17,r-cluster=2.0.3,r-codetools=0.2_14,r-foreign=0.8_66,r-futile.options=1.0.0,r-kernsmooth=2.23_15,r-lambda.r=1.1.7,r-lattice=0.20_33,r-mass=7.3_45,r-nnet=7.3_11,r-rpart=4.1_10,r-snow=0.4_1,r-spatial=7.3_11,r-survival=2.38_3,r-catools=1.17.1,r-class=7.3_14,r-futile.logger=1.4.1,r-matrix=1.2_2,r-nlme=3.1_122,bioconductor-biocparallel=1.4.3,r-mgcv=1.8_9,bioconductor-rsamtools=1.22.0,r-recommended=3.2.2,r-spp=1.13
```

Comment:
Is the exact version of iuc and so on relevant for you? If not we usually just specify the top-level packages your are interested in and let conda figure out the rest and freeze the container.

Reply:
I find that very convenient as well. 
In theory I'm not interested in the iuc version, but here I want a conda environment and a container to be as close as possible. I started by specifying top level packages and noticed that when I came back some time later and made a new installation, some of the dependencies had been updated ==> I have to fix the version of those as well. This is the very over-exaggerated version based on me fixating everything. 

My procedure was:
1. install environment based on main software
1. Make explicit dump of installed recipes
1. convert to hash.tsv-line

I understand there is no guarantee the container and conda environment will be exactly the same. This was my naive approach and I'm willing to agree that I'm taking it too far... If there is something better I'll do it that way, but I would want to provide both container and environment.

Is there a way export the package versions installed in the biocontainer? (That way I could hide the nitpicking... :) )

Reply:
I'm not against this. I was just wondering as it looks overly complex :)
But if this is what the community needs, I'm all in :)

> Is there a way export the package versions installed in the biocontainer? (That way I could hide the nitpicking... :) )

Not really, we could look at the generated logs and get the pulled packages if this helps? We might need to increase the log level for this.

Reply:
It is overly complex :) The current docker/conda setup doesn't return the same result this is my idea to ensure consistency. As long as it is OK, this would be nice to have and I wouldn't need the logs.

A parallel question; this is now version two and knowing myself there might be a couple of iterations before I reach the goal. Can I build mulit-package-containers locally for testing? The travis-script makes me think of bioconda-utils...

Reply:
Its part of galaxy-lib, see here for some documentation: https://github.com/galaxyproject/galaxy-lib/blob/master/docs/topics/mulled.rst

Its just a `pip install galaxy-lib` away.

Reply:
Thank you! I will give it a try.

Reply:
oh oh... now I know how to specify specific builds of a version.... :)